src/static/js/event_animals.js

```diff
@@ -0,0 +1,320 @@
+                            tableform.table_remove_selected_from_json(table, controller.rows);
+                            tableform.table_update(table);
+                        } 
+                    },
+                    { id: "animalarrived", text: _("Arrived"), icon: "complete", tooltip: _(""), enabled: "multi", perm: "cea",
+                        click: async function() { 
+                            await tableform.show_okcancel_dialog("#dialog-arrived", _("Ok"));
+                            tableform.buttons_default_state(buttons);
+                            let ids = tableform.table_ids(table);
+                            await common.ajax_post("event_animals", "mode=arrived&ids=" + ids);
+                            $.each(controller.rows, function(i, v) {
+                                if (tableform.table_id_selected(v.ID)) {
+                                    if(!v.ARRIVALDATE) {
+                                        v.ARRIVALDATE = format.date_now_iso();
+                                    }
+                                }
+                            });
+                            tableform.table_update(table);
+                        } 
+                    },
+                    { id: "animalendactivefoster", text: _("End active foster"), tooltip: _("Set current foster movement return date to event start or current time, whichever is later"), enabled: "multi", perm: "cea",
+                        click: async function() {
+                            await tableform.show_okcancel_dialog("#dialog-endactivefoster", _("Ok"));
+                            tableform.buttons_default_state(buttons);
+                            let ids = tableform.table_ids(table);
+                            await common.ajax_post("event_animals", "mode=endactivefoster&ids=" + ids);
+                            $.each(controller.rows, function(i, v) {
+                                if (tableform.table_id_selected(v.ID)) {
+                                    if(!v.LASTFOSTERERRETURNDATE) {
+                                        v.LASTFOSTERERRETURNDATE = format.date_now_iso();
+                                    }
+                                }
+                            });
+                            tableform.table_update(table);
+                        } 
+                    },
+                    { id: "filter", type: "dropdownfilter", 
+                     options: [ "all|" + _("All"), "arrived|" + _("Arrived"), 
+                        "noshow|" + _("No show"), "neednewfoster|" + _("Need new foster"), 
+                        "dontneednewfoster|" + _("Don't need new foster"),
+                        "adopted|" + _("Adopted"), 
+                        "notadopted|" + _("Not adopted") ],
+                     click: function(selval) {
+                        common.route(controller.name + "?id=" + controller.event.ID + "&filter=" + selval);
+                        common.route_reload();
```

Comment:
You don't need to do route_reload here, just setting a new route with the updated filter as your first line does is all you need. I think this is why your filter isn't working.

Reply:
I tried without route_reload and it didn't work... adding route_reload didn't help either :)

Reply:
when I drop route_reload, I see a request being made with the following payload:
![image](https://user-images.githubusercontent.com/1950417/218256644-2bdb3267-2744-4862-8368-81a498beb2e3.png)


while the browser url shows as:
![image](https://user-images.githubusercontent.com/1950417/218256636-e637ee97-2256-4a5b-a71b-3de3d80e1161.png)

If I then hit refresh, the payload shows everything that's in the url param list and the filtered list is returned correctly